client/extensions/woocommerce/components/query-settings-general/index.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { localize } from 'i18n-calypso';
+import { uniqueId } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { fetchSettingsGeneral } from 'woocommerce/state/sites/settings/general/actions';
+import { areSettingsGeneralLoaded, areSettingsGeneralLoadError } from 'woocommerce/state/sites/settings/general/selectors';
+import { errorNotice, removeNotice } from 'state/notices/actions';
+
+class QuerySettingsGeneral extends Component {
+	fetch( siteId ) {
+		this.props.actions.fetchSettingsGeneral( siteId );
+	}
+
+	componentWillMount() {
+		const { siteId, loaded } = this.props;
+
+		if ( siteId && ! loaded ) {
+			this.fetch( siteId );
+		}
+	}
+
+	componentWillReceiveProps( { siteId, loaded, error, translate } ) {
+		if ( siteId !== this.props.siteId && ! loaded ) {
+			// site ID changed, fetch new zones
```

Comment:
If `siteId` changes to a site for which nothing is loaded yet, then `loaded` will be `false`. If `loaded` is `true`, then that means that the settings are correctly fetched for `siteId`, so there's no need to fetch them again.